docs/developer/developer-manual.html

```diff
@@ -161,6 +161,24 @@ <h2 id="Build_tasks">Build tasks</h2>
 main directory or any subproject runs the <code>allTests</code> task only in the <code>framework</code> project.
 </p>
 
+<p>
+An iterative build-test-debug cycle where you assemble the Checker Framework
+and execute its tests can sometimes lead to a state where Gradle is not able to
+compile and build the system for no obvious reason.
+This is especially common when you kill tests before they fully finish, or kill
+compilation before it is fully complete, leading to a bad state or an
+inconsistent build cache.
+</p>
+
+<p>
+Below are some steps you might try to resolve this issue (in order):
+</p>
+<ul>
+  <li><code>./gradlew --no-build-cache assemble</code>: attempts to rebuild the Checker Framework without the build cache.
+  <li>Kill any Gradle daemons; run <code>./gradlew --status</code> to get a list of pids for Gradle.
+  <li>Delete the Gradle build cache, usually located at <code>~/.gradle/caches</code>. Note: this will delete the entire build
```

Comment:
Before doing this I would try `git clean -fdx` in the repo directory, to completely remove any files that are not in version control, including files covered by ignores.  (Of course you need to make sure you `git add` any new files you'd like to keep first.)  For me this command usually does the trick; I can't remember the last time I had to delete the Gradle build cache to fix a build.